pkg/cmd/codespace/ssh.go

```diff
@@ -216,25 +223,58 @@ func (a *App) SSH(ctx context.Context, sshArgs []string, opts sshOptions) (err e
 	}
 }
 
-func shouldUseAutomaticSSHKeys(args []string, opts sshOptions) bool {
-	if opts.profile != "" {
-		// The profile may specify the identity file so cautiously don't override anything with that option
-		return false
-	}
+func useAutomaticSSHKeys(
+	ctx context.Context,
+	sshContext ssh.Context,
+	apiClient apiClient,
+	args []string,
+	opts sshOptions,
+) (bool, error) {
+	customConfigPath := ""
+	for i := 0; i < len(args); i += 1 {
+		arg := args[i]
 
-	for _, arg := range args {
 		if arg == "-i" {
-			// User specified the identity file so it should exist
-			return false
+			// User specified the identity file so just trust it is correct
+			return false, nil
+		}
+
+		if arg == "-F" && i < len(args)-1 {
+			// ssh only pays attention to that last specified -F value, so it's correct to overwrite here
+			customConfigPath = args[i+1]
 		}
+	}
+
+	if automaticKeySSHKeysExist(sshContext) {
+		// If the automatic keys already exist, just use them
+		return true, nil
+	}
+
+	// If there is a public key uploaded which matches a configured
+	// private key, there is no need for automatic key generation
+	hasPublicKey, err := hasUploadedPublicKeyForConfig(ctx, sshContext, apiClient, customConfigPath, opts.profile)
+
+	return !hasPublicKey, err
+}
+
+func automaticKeySSHKeysExist(sshContext ssh.Context) bool {
+	publicKeys, err := sshContext.LocalPublicKeys()
+	if err != nil {
+		return false
+	}
```

Comment:
Return err as well?

Reply:
If this fails we don't really want to fail the whole command because the the api check could still succeed.

The error here is most likely that the `.ssh` directory doesn't exist, which is totally fine if the API check succeeds, and automatic key generation also handles this nicely. 

The other cases (`$HOME` not set, etc) would cause an error later on if/when we try to do the automatic keygen and the error there seems a bit more user friendly - `failed to generate ssh keys: $HOME not set` seems like a more expected error to me than `checking ssh key configuration: $HOME not set`.